### DISCUSSION TITLE
Email settings bug

### DISCUSSION THREAD

**Discussion by sbonaime** [requarks/wiki#3574] [Error / Bug Report]
When the TLS security is checked but username and password are left empty, sending a test email pop up this weird message

![Capture d’écran 2019-10-29 à 29-10-19 15 04 59](https://user-images.githubusercontent.com/750709/67774640-076d8a80-fa5e-11e9-9191-cb40a822ff24.png)

Maybe wikijs should check if TLS username and password are not empty when clicking on "Apply"

**Comment by robinhundt** [comment]
Also occurs when username and password are set.

**Comment by robinhundt** [comment]
Had a look at the E-Mail feature again.
For me it seems to have been the combination of a nonstandard port (587) and TLS activated that caused the issue.

**Comment by sbonaime** [comment]
In my case, I am using port 25.

**Comment by noahbjohnson** [comment]
![image](https://user-images.githubusercontent.com/14844541/73212688-5d60d580-4114-11ea-9b2e-8e3dd3fe0a54.png)

Getting this issue with a nonstandard combo like @robinhundt mentioned. 587 is the port that gmail uses for secure SMTP.

**Comment by ygingras** [comment]
I also get this error while using port 587 with both user and password set.  I'm running Wiki.js 2.1.113.

**Comment by Hindrik-H** [comment]
Hi everyone. 
Same error here, any news on it ?

**Comment by gauravpriyolkar** [comment]
+1

As another data point, get the same error when using Google's smtp relay configured to accept emails from allow-listed IPs, no password. Configured as follows:
https://support.google.com/a/answer/2956491#zippy=%2Creview-sending-limits-for-the-smtp-relay-service

**Reply by ddesmet** [reply]
+1

We're using the google workspace mail relay service with user authentication and IP whitelisting where TLS is enforced.

**Comment by 4pack** [comment]
It would seem like this is still an issue. I use a Postfix relay connected to Exchange Online. No credentials as well.

**Comment by tim-elmer** [comment]
Persists in 2.5.307, attempted all ports accepted by mailtrap; 465, 587; 2525. 25 doesn't display the error, but also doesn't work with TLS enabled (it does without, but I'm not leaving TLS off).

**Comment by jaronk2000** [comment]
I am on the alpha 3.0.0 version and I still getting this issue? Seems kind of crazy a 6 year long issue causing email to not function.
